radix-engine/src/transaction/executor.rs

```diff
@@ -156,85 +150,84 @@ impl<'l, L: Ledger> TransactionExecutor<'l, L> {
         &mut self,
         transaction: Transaction,
         trace: bool,
-    ) -> Result<Receipt, TransactionExecutionError> {
+    ) -> Result<Receipt, TransactionValidationError> {
```

Comment:
Seems to me that we should split the validation and the execution into two separate methods and have `run` accept `ValidatedTransaction`. This will be important when we need to pipeline transactions. For example, transaction validation can be parallelized and occur outside of the execution hotspots.